pkg/util/winutil/process.go

```diff
@@ -399,3 +399,35 @@ func getProcessStartTimeAsNs(pid uint64) (uint64, error) {
 	}
 	return uint64(creation.Nanoseconds()), nil
 }
+
+// KillProcess kills the process with the given PID, supplying the given return code
+func KillProcess(pid int, returnCode uint32) error {
+	/*
+	 * Open the process with PROCESS_TERMINATE rights.  This will fail
+	 * if the process is not owned by the current user, or the user does not
+	 * have admin rights
+	 */
+	h, err := windows.OpenProcess(windows.PROCESS_TERMINATE, false, uint32(pid))
+	if err != nil {
+		return fmt.Errorf("Error opening process %v", err)
+	}
+	/*
+	 * if the handle is successfully opened, must be closed to avoid handle leaks
+	 */
+	defer windows.Close(h)
+	/*
+	 * terminate the process; the process will exit with the given return code
+	 *
+	 * See https://learn.microsoft.com/en-us/windows/win32/api/processthreadsapi/nf-processthreadsapi-terminateprocess for
+	 * more information.
+	 *
+	 * A couple of notes:
+	 * - This function will return immediately.  The process itself will not be closed until all I/O is completed or cancelled
+	 * - We could block and wait for the process to terminate, but this is not done here.
+	 */
+	err = windows.TerminateProcess(h, returnCode)
+	if err != nil {
+		return fmt.Errorf("Error terminating process %v", err)
+	}
+	return nil
```

Comment:
Understanding this is a draft PR, we might want to consider this section of the docs for [TerminateProcess](https://learn.microsoft.com/en-us/windows/win32/api/processthreadsapi/nf-processthreadsapi-terminateprocess#remarks):
> When a process terminates itself, TerminateProcess stops execution of the calling thread and does not return. Otherwise, TerminateProcess is asynchronous; it initiates termination and returns immediately. If you need to be sure the process has terminated, call the [WaitForSingleObject](https://learn.microsoft.com/en-us/windows/win32/api/synchapi/nf-synchapi-waitforsingleobject) function with a handle to the process.

Since this is for CWS, we might want to be sure the process is actually terminated. Maybe a timeout or something otherwise since the function returns right away, we might think we terminated it when we didn't. Something to think about?

Reply:
Agreed, and noted 
https://github.com/DataDog/datadog-agent/pull/25311/files#diff-dc0948c7849fc8f9538fdd3eea18265e9e0088217942d99429a870f2892fd7b5R426

if we think we need some sort of completion notification (either make this call synchronous or have some sort of async callback mechanism when the process has actually been stopped), that would be additional scope.

Reply:
On linux we use the exit event with a timeout to report that the process has actually been killed. I'll try to use this with the common part on the agent handling the kill action

Reply:
@derekwbrown , whoops I missed the comment - my bad. In that case, I think that if this function returns `nil`, then to the caller that should be made clear that the process termination was successfully initiated, and not that it was actually terminated.

@safchain using the exit event is probably a good indicator that it was successfully terminated.

@derekwbrown @safchain - is this function planned to be called from security agent or system probe?

Reply:
This will only _work_ from system probe; if you call it without `admin` rights you can only termination your own processes.